### PR TITLE
Scatter extra name

### PR DESCRIPTION
**Pull request by chenjiandongx**
本次 PR 内容，

* Scatter 图新增 `extra_name` 参数，额外的数据项的名称，可以为每个数据点指定一个名称。

### CODE REVIEW

File: pyecharts/charts/scatter.py [hunk 1]
```diff
@@ -47,10 +56,8 @@ def __add(
         self._option.update(xAxis=xaxis, yAxis=yaxis)
         self._option.get("legend")[0].get("data").append(name)
 
-        if extra_data:
-            _data = [list(z) for z in zip(x_axis, y_axis, extra_data)]
-        else:
-            _data = [list(z) for z in zip(x_axis, y_axis)]
+        zip_lst = [i for i in (x_axis, y_axis, extra_data, extra_name) if i]
```
Comment (chfw) [inline pyecharts/charts/scatter.py]: what about x_axis=None, y_axis=None?

Comment (chenjiandongx) [reply pyecharts/charts/scatter.py]: 你这两个参数是用户必须提供的 所以不考虑这个问题

Comment (chfw) [reply pyecharts/charts/scatter.py]: 如何保证用户必须提供呢？

Comment (chfw) [reply pyecharts/charts/scatter.py]: x_axis=[], y_axis=[]?

Comment (chfw) [reply pyecharts/charts/scatter.py]: x_axis=['a'], y_axis=['1', '2']?

Comment (chenjiandongx) [reply pyecharts/charts/scatter.py]: 不觉得不提供 add 实际上是运行不了吗

File: pyecharts/charts/scatter.py [hunk 2]
```diff
@@ -47,10 +56,8 @@ def __add(
...
+        _data = [list(z) for z in zip(*zip_lst)]
```
Comment (chfw) [inline pyecharts/charts/scatter.py]: zip won't work as expected if the length of incoming list are not equal

File: pyecharts/charts/scatter.py (hunk 1 above)
Comment (chfw) [reply pyecharts/charts/scatter.py]: yep, I did realize x_axis and y_axis are positional arguments.

File: pyecharts/charts/scatter.py (hunk 2 above)
Comment (chenjiandongx) [reply pyecharts/charts/scatter.py]: 我觉得这点应该由用户保证 而不是我们

Comment (chfw) [reply pyecharts/charts/scatter.py]: a few assertions here will save time for our users: 10 seconds to realize what's wrong x 3424 (stars) = 34240 seconds = 9.5 hours

**Comment by chenjiandongx** [comment]
@chfw https://github.com/pyecharts/pyecharts/pull/652/commits/67fe21ee8b65d4679f1c1146dce29d8e5d386a43 here

**Comment by ColyFu** [comment]
Could you please add 'extra_name' to 3D Scatter? Thanks!

**Pull request merged**